libraries-3/src/main/java/com/baeldung/jnats/NatsClient.java

```diff
@@ -1,123 +1,142 @@
-
-            if (subscription != null) {
-                subscription.unsubscribe();
-            } else {
-                log.error("{} not found. Unable to unsubscribe.", topic);
-            }
-        } catch (IOException ioe) {
-            log.error("Error unsubscribing from {} ", topic, ioe);
+    public Subscription subscribeAsyncInQueueGroup(String subject, String queueGroup) {
+        Dispatcher dispatcher = natsConnection.createDispatcher();
+        Subscription subscription = dispatcher.subscribe(subject, queueGroup, msg -> log.info("Queue group subscription received message on {}", msg));
+        if (subscription == null) {
+            log.error("Error subscribing to {}", subject);
+            return null;
         }
+        dispatcherBySubscription.put(subscription, dispatcher);
+        return subscription;
     }
 
-    Message makeRequest(String topic, String request) {
-        try {
-            return natsConnection.request(topic, request.getBytes(), 100);
-        } catch (IOException | InterruptedException ioe) {
-            log.error("Error making request {} to {} ", topic, request, ioe);
+    public Subscription makeAsyncReplier(String subject, String reply) {
+        Dispatcher dispatcher = natsConnection.createDispatcher();
+        Subscription subscription = dispatcher.subscribe(subject, message -> publishMessage(message.getReplyTo(), reply));
+        if (subscription == null) {
+            log.error("Error making replier to {}", subject);
             return null;
         }
+        dispatcherBySubscription.put(subscription, dispatcher);
+        return subscription;
     }
 
-    void installReply(String topic, String reply) {
-        natsConnection.subscribe(topic, message -> {
-            try {
-                natsConnection.publish(message.getReplyTo(), reply.getBytes());
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-        });
+    public Subscription subscribeSync(String subject) {
+        return natsConnection.subscribe(subject);
+    }
+
+    public Subscription subscribeSyncInQueueGroup(String subject, String queueGroup) {
+        return natsConnection.subscribe(subject, queueGroup);
     }
 
-    SyncSubscription joinQueueGroup(String topic, String queue) {
-        return natsConnection.subscribe(topic, queue);
+    public CompletableFuture<Message> makeRequest(String subject, String request) {
+        return natsConnection.request(subject, request.getBytes());
+    }
+
+    public void unsubscribe(Subscription subscription) {
+        // async subscriptions made with a dispatcher must be unsubscribed through the dispatcher
+        // sync subscriptions can be unsubscribed directly through the subscription
+        Dispatcher dispatcher = dispatcherBySubscription.remove(subscription);
+        if (dispatcher == null) {
+            subscription.unsubscribe();
+        }
+        else {
```

Comment:
This looks like the Baeldung formatter is not being used, you can find it here:
https://github.com/Baeldung/dev-settings/tree/main/javaCodeStyle/intellij

Reply:
Fixed.